pyvista/core/filters/poly_data.py

```diff
@@ -2091,9 +2091,18 @@ def flip_normals(poly_data):
         if not poly_data.is_all_triangles():
             raise NotAllTrianglesError('Can only flip normals on an all triangle mesh.')
 
-        f = poly_data.faces.reshape((-1, 4))
-        f[:, 1:] = f[:, 1:][:, ::-1]
-        poly_data.faces = f
+        if _vtk.VTK9:
+            # use new connectivity API
+            faces = poly_data._connectivity_array.reshape(-1, 3)
+
+            # shuffle these in-place
+            # See: https://stackoverflow.com/a/33362288/3369879
+            faces[:, 0], faces[:, 2] = faces[:, 2], faces[:, 0].copy()
```

Comment:
Here's hoping the comment will be enough to prevent people from removing that `copy()` call...

Reply:
I think it will, especially now that it will be here twice.  Tests will fail without the copy.